testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1674,6 +1674,36 @@ public void testRenameTableAcrossSchema()
         assertFalse(getQueryRunner().tableExists(getSession(), renamedTable));
     }
 
+    @Test
+    public void testRenameTableUnqualifiedName()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_SCHEMA) && hasBehavior(SUPPORTS_CREATE_TABLE));
+
+        String sourceSchemaName = "test_source_schema_" + randomTableSuffix();
+        assertUpdate("CREATE SCHEMA " + sourceSchemaName);
+
+        String tableName = "test_rename_unqualified_name_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + sourceSchemaName + "." + tableName + " AS SELECT 123 x", 1);
+
+        String renamedTable = "test_rename_unqualified_name_new_" + randomTableSuffix();
+        if (!hasBehavior(SUPPORTS_RENAME_TABLE)) {
+            assertQueryFails("ALTER TABLE " + sourceSchemaName + "." + tableName + " RENAME TO " + sourceSchemaName + "." + renamedTable, "This connector does not support renaming tables");
+            assertUpdate("DROP TABLE " + sourceSchemaName + "." + tableName);
+            assertUpdate("DROP SCHEMA " + sourceSchemaName);
+            return;
+        }
+
+        assertUpdate("ALTER TABLE " + sourceSchemaName + "." + tableName + " RENAME TO " + renamedTable);
+        assertQuery("SELECT x FROM " + sourceSchemaName + "." + renamedTable, "VALUES 123");
+
+        assertUpdate("DROP TABLE " + sourceSchemaName + "." + renamedTable);
+
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
```

Comment:
this should be asserted after `assertUpdate("ALTER TABLE " +`. , not after `assertUpdate("DROP TABLE " renamed`